lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -459,6 +459,40 @@ function dosomething_user_user_pass_submit($form, &$form_state) {
 }
 
 /**
+ * Trigger events after a user successfully logs in.
+ *
+ * Implements hook_user_login.
+ */
+function dosomething_user_user_login(&$form_state, $account) {
+  if (!module_exists('dosomething_northstar')) return;
+
+  $northstar_user = null;
+
+  // Figure out whether user is logging in via email or mobile. Then, we'll
+  // send a request to Northstar's `user/verify` endpoint to see if Northstar
+  // is also able to successfully authenticate with the given credentials.
+  if ($account = dosomething_user_get_user_by_email($form_state['values']['name'])) {
+    $northstar_user = dosomething_northstar_verify_user([
+      'email' => $form_state['values']['name'],
+      'password' => $form_state['values']['pass'],
+    ]);
+
+  }
+  elseif ($number = $form_state['values']['name']) {
+    if ($account = dosomething_user_get_user_by_mobile($number)) {
```

Comment:
The `$account` parameter given in the hook only includes the hashed password, and doesn't specify which "username" credential the user tried to log in with.